### PR TITLE
Add support for --no-new-privs

### PR DESCRIPTION
**Pull request by baude**
This replaces Dan's original PR #218.  As of now, no changes other than a rebase were required.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by mheon** [comment]
I think we can actually remove it from the DB altogether, given that it's in the spec

File: cmd/podman/spec.go [hunk 1]
```diff
@@ -645,6 +645,7 @@ func (c *createConfig) GetContainerCreateOptions() ([]libpod.CtrCreateOption, er
 	}
 
 	options = append(options, libpod.WithPrivileged(c.Privileged))
+	options = append(options, libpod.WithNoNewPrivs(c.NoNewPrivs))
```
Comment (mheon) [inline cmd/podman/spec.go]: Drop this

File: libpod/options.go [hunk 2]
```diff
@@ -272,6 +272,18 @@ func WithPrivileged(privileged bool) CtrCreateOption {
 	}
 }
 
+// WithNoNewPrivs sets the noNewPrivs flag in the container runtime
```
Comment (mheon) [inline libpod/options.go]: Remove this function

File: cmd/podman/spec.go (hunk 1 above)
Comment (baude) [reply cmd/podman/spec.go]: fixed

File: libpod/options.go (hunk 2 above)
Comment (baude) [reply libpod/options.go]: fixed

**Comment by baude** [comment]
@mheon @rhatdan im almost dead sure this is a selinux error.  @rhatdan did you say you were expecting a new selinux package for this too?

**Comment by rhatdan** [comment]
@baude are you able to get this to fail locally? Or just in the testing?  Is there a way I can get onto a testing machine where this is happening.

**Comment by jlebon** [comment]
The `FAH27` one has:

> "kernel": "4.14.18-300.fc27.x86_64",

While the `Fedora fedora/27/cloud Podman` one has:

> "kernel": "4.13.9-300.fc27.x86_64",

As opposed to AH, Fedora usually does not update cloud images after release. You'd have to update the kernel and reboot as a separate step before starting the tests. Which is tricky of course, since the code is running *on* the host. One way around this is to define a `cluster` instead:

```yaml
cluster:
    hosts:
        - name: node
          distro: fedora/27/cloud
          specs:
              ram: 8192
              cpus: 4
    container:
        image: registry.fedoraproject.org/fedora:27
```

So then your code will run in the container, so you can e.g. `ssh node dnf update -y kernel` and then `ssh node reboot || :` or something along those lines.

**Comment by jlebon** [comment]
Follow-up: you can now use `fedora/27/cloud/pungi`, which is updated at least once a day and should have the latest kernel (see https://github.com/projectatomic/papr/pull/87). But **caveat emptor**, there's no guarantee these images are going to be as stable as `fedora/27/cloud`.

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
LGTM

**Comment by baude** [comment]
ill add in some conditional check to the test looking for 4.14 or newer

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**